src/main/java/com/github/alinaberlin/security/components/JWTEncoderDecoder.java

```diff
@@ -0,0 +1,87 @@
+package com.github.alinaberlin.security.components;
+
+import com.github.alinaberlin.security.models.User;
+import io.jsonwebtoken.*;
+import jakarta.servlet.http.HttpServletRequest;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.stereotype.Component;
+
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+@Component
+public class JWTEncoderDecoder {
```

Comment:
I recommend creating an interface that this class would implement and override the functions that you have so it is easier to replace it in the future if requirements change.